### PR TITLE
libsbml: update to version 5.16.0

### PR DESCRIPTION
**Pull request by funasoul**
* update to version 5.16.0
* use cmake to build libSBML
* add maintainer
* add SBML Level 3 extension packages
* add Java, Python35 and Python36 bindings
* add cppnamespace, expat and libxml2 variants

###### Type(s)

- [x] enhancement

###### Tested on

macOS 10.13.3 17D47
Xcode 9.2 9C40b

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?

- [x] checked your Portfile with `port lint`?
- [x] tried a full install with `sudo port -vst install`? (`sudo port -vs install` worked, but `sudo port -vst install` failed. see https://trac.macports.org/ticket/55575)
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: science/libsbml/Portfile [hunk 1]
```diff
@@ -1,12 +1,15 @@
 # -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
 
 PortSystem          1.0
+PortGroup           cmake 1.0
+
+cmake.out_of_source yes
```
Comment (mf2k) [inline science/libsbml/Portfile]: Please test and use the cmake 1.1 portgroup instead. See https://trac.macports.org/ticket/47197#comment:63.

File: science/libsbml/Portfile [hunk 2]
```diff
@@ -18,32 +21,117 @@ long_description    LibSBML is a free, open-source programming library to \
...
+    configure.args-append   -DWITH_CPP_NAMESPACE:BOOL=ON
+}
+
+variant arrays description {Enable libSBML support for the SBML Level 3 Arrays and Sets ('arrays')} {
+    configure.args-append   -DENABLE_ARRAYS:BOOL=ON
+}
+
+variant comp description {Enable libSBML support for the SBML Level 3 Hierarchical Model ('comp')} {
+    configure.args-append   -DENABLE_COMP:BOOL=ON
+}
```
Comment (mf2k) [inline science/libsbml/Portfile]: It's nitpicky, but please keep the variants sorted by name.

Comment (funasoul) [reply science/libsbml/Portfile]: Thank you for your comment.
I've changed the Portfile to use cmake 1.1 and sorted the variants by name.

File: science/libsbml/Portfile [hunk 3]
```diff
@@ -18,32 +19,117 @@ long_description    LibSBML is a free, open-source programming library to \
                     rather a library you can embed in your own applications.
 
 homepage            http://sbml.org/Software/libSBML
-master_sites        sourceforge:project/sbml/libsbml/${version}/stable
+master_sites        sourceforge:project/sbml/libsbml/${version}/experimental/source
```
Comment (mf2k) [inline science/libsbml/Portfile]: Why did you change the master_sites to point to the experimental directory? 5.16.0 is the latest stable version and is in the stable directory like original master_sites. Please restore the master_sites line to the original value.

Comment (funasoul) [reply science/libsbml/Portfile]: Thank you for the comment.

As you can see in https://sourceforge.net/projects/sbml/files/libsbml/5.16.0/, the difference between `experimental` and `stable` is whether it includes the code of beta packages (SBML extensions) or not (which means, there is no difference in SBML-core).

The reason why I have used the experimental code is that I wanted to support SBML extension packages (arrays, distrib, dyn, render, req, and spatial) in this port. The extension packages are not installed by default unless the user adds some variants (+arrays +distrib +dyn etc.).

Moreover, the code between experimental and stable is identical except the above extension packages. This means that there is no difference in generated libSBML if no variants on the extension is added. Here are the md5 value of `libsbml.5.16.0.dylib` on both stable and experimental. Both libsbml.5.16.0.dylib was generated with the same cmake option (without any variants).

```
% md5 stable/build/src/libsbml.5.16.0.dylib experimental/build/src/libsbml.5.16.0.dylib
MD5 (stable/build/src/libsbml.5.16.0.dylib) = d67fcbf2229c61d9b12d0cb875554717
MD5 (experimental/build/src/libsbml.5.16.0.dylib) = d67fcbf2229c61d9b12d0cb875554717
```

Here is the command line I used to check the difference between stable and experimental, and the diff file is also attached to this message.

```
% diff -urBb libSBML-5.16.0-Source-experimental/src libSBML-5.16.0-Source-stable/src > src.diff
```

Please let me know if you have a further question about it.

[src.diff.gz](https://github.com/macports/macports-ports/files/1737743/src.diff.gz)

Comment (pmetzger) [reply science/libsbml/Portfile]: @mf2k Opinions? Should we pull at this point?

Comment (mf2k) [reply science/libsbml/Portfile]: Yes, I'm good with this.

File: science/libsbml/Portfile [hunk 4]
```diff
@@ -1,12 +1,13 @@
...
 PortSystem          1.0
+PortGroup           cmake 1.1
 
 name                libsbml
-version             5.13.0
+version             5.16.0
 categories          science
 platforms           darwin
-maintainers         nomaintainer
+maintainers         @funasoul openmaintainer
```
Comment (pmetzger) [inline science/libsbml/Portfile]: Thanks for taking over as the maintainer. You should include both your github account name and your obfuscated email address so that you can get notifications about things like failures of automatic builds. Instructions are in the macports documentation.

**Comment by pmetzger** [comment]
I've requested a small change to your maintainer line. Also, please squash all your commits into one commit so that when we pull up the changes there is only one commit to the MacPorts repository. When this is done we can pull up the request.

File: science/libsbml/Portfile (hunk 4 above)
Comment (funasoul) [reply science/libsbml/Portfile]: Thanks, I've added my email address and squashed my commits into one commit.

File: science/libsbml/Portfile [hunk 5]
```diff
@@ -1,12 +1,13 @@
...
 PortSystem          1.0
+PortGroup           cmake 1.1
 
 name                libsbml
-version             5.13.0
+version             5.16.0
 categories          science
 platforms           darwin
-maintainers         nomaintainer
+maintainers         @funasoul gmail.com:funasoul openmaintainer
```
Comment (pmetzger) [inline science/libsbml/Portfile]: Sorry to keep picking at details. This should be 

`maintainers         {@funasoul gmail.com:funasoul} openmaintainer`

And thank you for your patience.

Comment (funasoul) [reply science/libsbml/Portfile]: Thanks for pointing out the issue. I've committed the fix.

**Comment by pmetzger** [comment]
Merged! Thank you for your contribution to MacPorts!
